command/certificate/p12.go

```diff
@@ -0,0 +1,140 @@
+package certificate
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/command"
+	"github.com/smallstep/cli/crypto/pemutil"
+	"github.com/smallstep/cli/errs"
+	"github.com/smallstep/cli/flags"
+	"github.com/smallstep/cli/ui"
+	"github.com/smallstep/cli/utils"
+	"github.com/urfave/cli"
+
+	"software.sslmate.com/src/go-pkcs12"
+)
+
+func p12Command() cli.Command {
+	return cli.Command{
+		Name:      "p12",
+		Action:    command.ActionFunc(p12Action),
+		Usage:     `package a certificate and keys into a .p12 file`,
+		UsageText: `step certificate p12 <p12_path> [<crt_path>] [<key_path>] [**--ca**=<ca_crt_path>] [**--password**=<password>]`,
+		Description: `**step certificate p12** creates a .p12 (PFX / PKCS12)
+		file containing certificates and keys. This can then be used to import
+		into Windows / Firefox / Java applications.
+
+## EXIT CODES
+
+This command returns 0 on success and \>0 if any error occurs.
+
+## EXAMPLES
+
+Package a certificate and private key together:
+
+'''
+$ step certificate p12 foo.p12 foo.crt foo.key
+'''
+
+Package a certificate and private key together, and include an intermediate certificate:
+
+'''
+$ step certificate p12 foo.p12 foo.crt foo.key --ca intermediate.crt
+'''
+
+Package a CA certificate into a "trust store" for Java applications
+
+'''
+$ step certificate p12 trust.p12 --ca ca.crt
+'''
+`,
+		Flags: []cli.Flag{
+			cli.StringSliceFlag{
+				Name: "ca",
+				Usage: `Add a CA or intermediate certificate to the .p12 file. Use the '--ca'
+flag multiple times to add multiple CAs or intermediates.`,
+			},
+			cli.StringFlag{
+				Name:  "password",
+				Usage: `Set the password to encrypt the .p12 file.`,
+			},
```

Comment:
For consistency with other commands, it makes more sense to use `--password-file`, and then just use `utils.ReadStringPasswordFromFile(filename string) (string, error)`.